types/bitcoinjs-lib/index.d.ts

```diff
@@ -232,172 +352,969 @@ export const opcodes: {
+
     function compile(chunks: any): any;
-    function decompile(buffer: any): any;
+
+    function decompile(buffer: Buffer): any;
+
     function fromASM(asm: any): any;
-    function isCanonicalPubKey(buffer: any): any;
-    function isCanonicalSignature(buffer: any): any;
+
+    function isCanonicalPubKey(buffer: Buffer): any;
+
+    function isCanonicalSignature(buffer: Buffer): any;
+
     function isDefinedHashType(hashType: any): any;
-    function isMultisigInput(script: any, allowIncomplete: any): any;
-    function isMultisigOutput(script: any): any;
-    function isNullDataOutput(script: any): any;
-    function isPubKeyHashInput(script: any): any;
-    function isPubKeyHashOutput(script: any): any;
-    function isPubKeyInput(script: any): any;
-    function isPubKeyOutput(script: any): any;
-    function isScriptHashInput(script: any, allowIncomplete: any): any;
-    function isScriptHashOutput(script: any): any;
-    function isWitnessPubKeyHashOutput(script: any): any;
-    function isWitnessScriptHashOutput(script: any): any;
-    function multisigInput(signatures: any, scriptPubKey: any): any;
-    function multisigOutput(m: any, pubKeys: any, ...args: any[]): any;
-    function nullDataOutput(data: any): any;
-    function pubKeyHashInput(signature: any, pubKey: any, ...args: any[]): any;
-    function pubKeyHashOutput(pubKeyHash: any): any;
-    function pubKeyInput(signature: any): any;
-    function pubKeyOutput(pubKey: any): any;
-    function scriptHashInput(scriptSig: any, scriptPubKey: any): any;
-    function scriptHashOutput(scriptHash: any): any;
+
+    function isPushOnly(value: any): any;
+
     function toASM(chunks: any): any;
-    function witnessPubKeyHashOutput(pubKeyHash: any): any;
-    function witnessScriptHashInput(scriptSig: any, scriptPubKey: any): any;
-    function witnessScriptHashOutput(scriptHash: any): any;
+
+    function toStack(chunks: any): any;
+
+    namespace classifyInput {
+        const prototype: {
+        };
+
+    }
+
+    namespace classifyOutput {
+        const prototype: {
+        };
+
+    }
+
+    namespace classifyWitness {
+        const prototype: {
+        };
+
+    }
+
+    namespace compile {
+        const prototype: {
+        };
+
+    }
+
+    namespace decompile {
+        const prototype: {
+        };
+
+    }
+
+    namespace fromASM {
+        const prototype: {
+        };
+
+    }
+
+    namespace isCanonicalPubKey {
+        const prototype: {
+        };
+
+    }
+
+    namespace isCanonicalSignature {
+        const prototype: {
+        };
+
+    }
+
+    namespace isDefinedHashType {
+        const prototype: {
+        };
+
+    }
+
+    namespace isPushOnly {
+        const prototype: {
+        };
+
+    }
+
     namespace number {
-        function decode(buffer: any, maxLength: any, minimal: any): any;
-        function encode(number: any): any;
+        function decode(buffer: Buffer, maxLength: any, minimal: any): any;
```

Comment:
`maxLength` has type `number`. 
`minimal` has type `boolean` and is optional.
Returns `number`.